### PR TITLE
completed task 6 (kinda can't believe it)

### PR DESCRIPTION
**Pull request by usmosis**

### CODE REVIEW

**Comment by usmosis** [comment]
sorry i left a console log in my test file... my bad

File: __tests__/app.test.js [hunk 1]
```diff
@@ -93,4 +93,42 @@ describe('GET: /api/articles', () => {
...
+        })
+    })
+    it("GET:400 sends an appropriate error message when given an invalid article_id", () => {
+        return request(app).get('/api/articles/dog/comments')
+        .expect(400)
+        .then(({body}) => {
+            const {msg} = body;
+            expect(msg).toBe("Bad request")
+        })
+    })
```
Comment (nickkagame) [inline __tests__/app.test.js:140]: We could do with another test here for an article that exists but has no comments!   This shouldn't be an error.  Should return a 200, but with no content!

File: models/comments.models.js [hunk 2]
```diff
@@ -1,2 +1,13 @@
 const db = require('../db/connection')
 
+exports.selectCommentsByArticleId = (article_id) => {
+    return db.query("SELECT * FROM comments JOIN articles ON articles.article_id = comments.article_id WHERE comments.article_id = $1;", [article_id])
+    .then((results) => {
+        if(!results.rows[0]){
+            return Promise.reject({status: 404, msg: "Not found"})
```
Comment (nickkagame) [inline models/comments.models.js]: what will happen what we invoke this model with article_id for an article that has no comments?  We need to check that the article we want comments for exists, and only send an error if it doesn't.  The lecture today covered this if you were around!

**Review by nickkagame — CHANGES_REQUESTED** [review]
Nice work overall.  Just need to handle the case of making a request for comments on article that has no comments.  In this case you wouldn't want to throw and error!

File: controllers/comments.controllers.js [hunk 3]
```diff
@@ -4,15 +4,18 @@ const {selectArticleById} = require('../models/articles.models')
 exports.sendCommentsByArticleId = (req, res, next) => {
 
     const {article_id} = req.params;
+    const promises = [selectArticleById(article_id)]
 
-    selectArticleById(article_id)
-    .then(() => {
-        return selectCommentsByArticleId(article_id)
+    if(article_id){
+        promises.push(selectCommentsByArticleId(article_id))
```
Comment (nickkagame) [inline controllers/comments.controllers.js:10]: Nice use of Promise.all!  I would remove (line 9 -11) the step of checking if(article_id) and pushing the promises array.   You can just put the selectCommentsByArticleID function in the promise array to start with.  You are getting comments by article ID. If article_id is undefined, you'll just get a psql error that your error handlers will get anyway anyway!  This function should not work without an article_id

**Review by nickkagame — APPROVED** [review]
Nice! All good.  Just a little comment about 3 unnecessary lines of code in there. Merge when ready.

**Pull request merged**